### ISSUE TITLE
Fix `function-url-quotes` autofix for double-slash comments in SCSS maps

### ISSUE THREAD

**Issue by jgerigmeyer** [stylelint/stylelint#6741]
### What steps are needed to reproduce the bug?

```scss
$test-map: (
  // test comment
  foo: 'bar'
);
```

Might be a bug with the `postcss-scss` custom syntax imported by stylelint-config-recommended-scss?

### What Stylelint configuration is needed to reproduce the bug?

```yaml
extends:
  - stylelint-config-recommended-scss

rules:
  function-url-quotes: always
```

or

```yaml
extends:
  - stylelint-config-recommended-scss

rules:
  function-url-quotes: never
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkAXGBnFBaAtgQwAd4ACACgB0A7Y4geluLU2MhxxkpSpoDMIISAcgBGeAE6CqASgDcVEABoQPAJYAbGADk87BCBgAPHQQ0A6dGHTpF4CJVUBzPYbSUo6eN2JZimAJ4aaiqcWJD2Kg5YYjCs7G4wUFgWVlRUYgCuGh5ePOmUYCgqdljpYmpYAI7pEEwkeGoA7nh+6PJKYY4AYhBi+Ch6fjpqNrAE1ojAXuQg-oHBXAjE02p4TAsKUzMoATBBIR0RUTEQbBywicmti8urGAtUAL4gD0A

### Which version of Stylelint are you using?

`15.3.0`

### What did you expect to happen?

No change to original code:

```scss
$test-map: (
  // test comment
  foo: 'bar'
);
```

### What actually happened?

```scss
$test-map: (
  /* test comment*/
  foo: 'bar'
);
```

### Does the bug relate to non-standard syntax?

SCSS maps

**Comment by ybiquitous** [comment]
@jgerigmeyer Thanks for the report. We can reproduce the bug with the demo.

When I looked into the `function-url-quotes` implementation, perhaps the following function seems a problem:

https://github.com/stylelint/stylelint/blob/fe4a86f8e341ef9f50dd6163f312da73f004ea02/lib/rules/function-url-quotes/index.js#L58-L68

The `getDeclarationValue()` utility checks `.raws.value.raw`, while `postcss-scss` produces `.raws.value.scss`:

https://github.com/stylelint/stylelint/blob/fe4a86f8e341ef9f50dd6163f312da73f004ea02/lib/utils/getDeclarationValue.js#L7-L11

```console
$ node -pe 'require("postcss-scss").parse("$a: (\n// foo\n)").nodes[0].raws.value'
{ value: '(\n\n)', raw: '(\n/* foo*/\n)', scss: '(\n// foo\n)' }
```

The simplest way to resolve this is to ignore such non-standard CSS by using the [`isStandardSyntaxDeclaration()`](https://github.com/stylelint/stylelint/blob/fe4a86f8e341ef9f50dd6163f312da73f004ea02/lib/utils/isStandardSyntaxDeclaration.js#L11) utility like this:

```js
function checkDeclParams(decl) {
	if (!isStandardSyntaxDeclaration(decl)) return;

	// ...
}
```

Note that Stylelint core processes only *standard* CSS. See also the [doc](https://stylelint.io/developer-guide/rules/).

---

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jgerigmeyer** [comment]
@ybiquitous Thanks! Fix submitted in #6745 🚀

**Comment by PurpleTape** [comment]
Hello! I noticed the same problem with comments when using the SCSS 'with()' construct:
```scss
// module.scss
$foo: null !default;
```
usage:
```scss
@forward 'module' with (
   // comment
   $foo: 'bar'
)
```

replaced to:
```scss
@forward 'module' with (
   /* comment*/
   $foo: 'bar'
)
```
